src/func.d

```diff
@@ -1276,7 +1261,8 @@ extern (C++) class FuncDeclaration : Declaration
          */
         TemplateInstance ti;
         if (fbody &&
-            (isFuncLiteralDeclaration() || (storage_class & STCinference) || (inferRetType && !isCtorDeclaration()) || isInstantiated() && !isVirtualMethod() &&
+            (sc.func || isFuncLiteralDeclaration() || (storage_class & STCinference) || (inferRetType && !isCtorDeclaration()) ||
```

Comment:
Added sc.func here, because nested functions with bodies should have their attributes inferred unless overridden.